views/js/runner/provider/qti.js

```diff
@@ -145,24 +145,34 @@ define([
          * Clean up stuffs
          */
         clear : function(elt, done){
-            if(this._item){
+            var self = this;
 
-                _.invoke(this._item.getInteractions(), 'clear');
-                this._item.clear();
+            if (self._item) {
+                Promise.all(this._item.getInteractions().map(function(interaction) {
+                    return interaction.clear();
+                })).then(
+                    function() {
+                        self._item.clear();
+
+                        $(elt).off('responseChange')
+                            .off('endattempt')
+                            .off('themechange')
+                            .off('feedback')
+                            .empty();
 
-                $(elt).off('responseChange')
-                      .off('endattempt')
-                      .off('themechange')
-                      .off('feedback')
-                      .empty();
+                        if(self._renderer){
+                            self._renderer.unload();
+                        }
 
-                if(this._renderer){
-                    this._renderer.unload();
-                }
+                        self._item = null;
 
-                this._item = null;
+                        done();
+                    },
+                    function() {
+                        throw new Error('Interaction clear failed');
```

Comment:
Instead of throwing which will stop the current context, please trigger an error.